recipes/MOABB/MotorImagery_decoding/BNCI2014001/MOABB_dataio_iterators.py

```diff
@@ -0,0 +1,409 @@
+import numpy as np
+from moabb.datasets.base import BaseDataset
+from moabb.paradigms.base import BaseParadigm
+from sklearn.model_selection import StratifiedKFold
+from itertools import islice
+import torch
+from torch.utils.data import TensorDataset, DataLoader
+import os
+
+
+def standardize(x, m, s):
+    """This function standardizes input EEG signals (x) using mean value (m)
+    and standard deviation (s)."""
+    demeaned = x - m.reshape((1, m.shape[0], 1))
+    standardized = demeaned / (1e-14 + s.reshape((1, s.shape[0], 1)))
+    return standardized
+
+
+def nth(iterable, n, default=None):
```

Comment:
I think this is used on the StratifiedKFold splits...
don't we want to split the data such that we don't have the same sessions of the same subject on train and test?
I think there's a data leakage problem if we do like this. In fact, you already "know" your subject from the train set.
Maybe this is implemented somewhere else and I am just missing it!